### PR TITLE
[backend/frontend] Export investigation in JSON as STIX report (#1374)

### LINKED ISSUE #1374: Export investigation in JSON or CSV format
Please replace every line in curly brackets { like this } with appropriate answers, and remove this line.

## Problem to Solve

It would be useful to be able to export investigations made from the investigation panel into JSON (and maybe CSV), in order to be able to share it with others. 

The envisioned workflow is the following : 
1/ several reports are integrated to OpenCTI
2/ An analyst investigate the knowledge from the investigation tab
3/ Based on the result of the investigation, the analyst write a short analysis/commentary, and join the JSON of his/her OpenCTi investigation in order to share it with partners, constituents, clients etc. 
4/ The JSON can be directly integrated in tools of the partners.

## Current Workaround

Make an investigation from the knowledge tab of a report in order to be able to export it.

## Proposed Solution
The export could be a button at the top right, or a dedicated tab for files, just like it exists in other spaces today.
Only the objects displayed on the graph should be exported, with the displayed relations. This should take into account active filters. 
Attributes and parameters of the object should be exported (source, dates, confidence level).
A useful option could be to be able to add for each object and each relation the list of related reports.

### PR DESCRIPTION
**Pull request by SouadHadjiat**
### Proposed changes

* Add toStixReportBundle query, which returns the bundle of a workspace (only for investigation type) with all the investigation objects (stix entities and relationships) and creates a new report containing all these objects.
* Add a button "Download as STIX report" in investigation header : calls toStixReportBundle and creates a file with the bundle to download.

With this file, you can import it as a workbench, then validate it to create the report generated from the investigation.

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/1374

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Archidoit** [comment]
Tested, everything is Ok :) 
Just a question to be sure: the relations of type refs are not exported? (only relationships and sightings are)
Ex: in the capture below, only 3 relationships are exported and found in the generated report (the 2 'targets' and 1 'uses')
image

**Comment by SouadHadjiat** [comment]
> Tested, everything is Ok :) Just a question to be sure: the relations of type refs are not exported? (only relationships and sightings are) Ex: in the capture below, only 3 relationships are exported and found in the generated report (the 2 'targets' and 1 'uses') image

Yes, only stix relationships are exported, and the "refs" are not stix relationships. I exported first all relationships but I had an error (because refs couldn't be converted in stix) and @RomuDeuxfois helped me undertstand it.

Thank you for your tests @Archidoit !

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts]: You can use the constant
ENTITY_TYPE_CONTAINER_REPORT.toLowerCase()

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts]: Is it really an export?

I would rather have thought for toStixBundleReport

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts:87]: We don't have an enum to handle it ?

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.jsx
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.jsx]: It's maybe better for the name to be close to the actual export file (connector side) :
image

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts:94]: Can overlap an existing report at import (maybe is a KO limit)

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts:107]: @richard-julien Do we want to export a json rather than a string so it can be used directly via the API?

**Comment by RomuDeuxfois** [comment]
Tested ✅

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts:87]: I didn't find it, and I thought about creating one, but for now it's the only place we use it, so it didn't seem very useful to create an enum for one condition.

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts:94]: This was requested in the brainstorm to name the report after the investigation

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts]: I could even use this method : `convertTypeToStixType(ENTITY_TYPE_CONTAINER_REPORT)`

**Comment by marieflorescontact** [comment]
Tested, looks good

File: opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts
Comment (RomuDeuxfois) [reply opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts:107]: @Kedae maybe an opinion on that ?

**Comment by SouadHadjiat** [comment]
PR status : ready for more reviews / waiting for approval

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**
